crates/re_format/src/arrow.rs

```diff
@@ -3,22 +3,102 @@
 use std::fmt::Formatter;
 
 use arrow2::{
-    array::{get_display, Array},
+    array::{get_display, Array, ListArray, PrimitiveArray, StructArray},
     datatypes::{DataType, IntervalUnit, TimeUnit},
 };
 use comfy_table::{presets, Cell, Table};
 
+// ---
+
+// TODO(#1775): Registering custom formatters should be done from other crates:
+// A) Because `re_format` cannot depend on other crates (cyclic deps)
+// B) Because how to deserialize and inspect some type is a private implementation detail of that
+//    type, re_format shouldn't know how to deserialize a TUID...
+
+type CustomFormatter<'a, F> = Box<dyn Fn(&mut F, usize) -> std::fmt::Result + 'a>;
+
+pub fn get_custom_display<'a, F: std::fmt::Write + 'a>(
+    _column_name: &'a str,
+    array: &'a dyn Array,
+    null: &'static str,
+) -> CustomFormatter<'a, F> {
+    // NOTE: If the top-level array is a list, it's probably not the type we're looking for: we're
+    // interested in the type of the array that's underneath.
+    let datatype = (|| match array.data_type().to_logical_type() {
+        DataType::List(_) => array
+            .as_any()
+            .downcast_ref::<ListArray<i32>>()?
+            .iter()
+            .next()?
+            .map(|array| array.data_type().clone()),
+        _ => Some(array.data_type().clone()),
+    })()
+    .unwrap();
+
+    if let DataType::Extension(name, _, _) = datatype {
+        match name.as_str() {
+            // TODO(#1775): This should be registered dynamically.
+            "rerun.tuid" => Box::new(|w, index| {
+                if let Some(tuid) = parse_tuid(array, index) {
+                    w.write_fmt(format_args!("{tuid:032X}"))
+                } else {
+                    w.write_str("[]")
+                }
+            }),
+            _ => get_display(array, null),
+        }
+    } else {
+        get_display(array, null)
+    }
+}
+
+// TODO(#1775): This should be defined and registered by the `re_tuid` crate.
+fn parse_tuid(array: &dyn Array, index: usize) -> Option<u128> {
+    let (array, index) = match array.data_type().to_logical_type() {
+        // Legacy MsgId lists: just grab the first value, they're all identical
+        DataType::List(_) => (
+            array
+                .as_any()
+                .downcast_ref::<ListArray<i32>>()?
+                .value(index),
+            0,
+        ),
+        // New control columns: it's not a list to begin with!
+        _ => (array.to_boxed(), index),
+    };
+    let array = array.as_any().downcast_ref::<StructArray>()?;
+
+    let time_ns = array
+        .values()
+        .first()?
+        .as_any()
+        .downcast_ref::<PrimitiveArray<u64>>()?
+        .values()
+        .get(index)?;
+    let inc = array
+        .values()
+        .get(1)?
+        .as_any()
+        .downcast_ref::<PrimitiveArray<u64>>()?
+        .values()
+        .get(index)?;
+
+    Some(((*time_ns as u128) << 64) | (*inc as u128))
+}
```

Comment:
This is extremely hacky and dependent on the internals of `Tuid`. If those change, this will silently break.

Cannot this be done via `arrow2_convert` instead?

Reply:
> This is extremely hacky and dependent on the internals of `Tuid`. If those change, this will silently break.

It is hacky but it won't silently break: it'll print an error value (which I've just changed from `[]`, which was certainly pretty bad, to `<ERR>`):
![image](https://user-images.githubusercontent.com/2910679/230311576-c9d06c7d-2aa0-4cd7-bacb-c5032fed08a4.png)
 
> Cannot this be done via `arrow2_convert` instead?

We cannot depend on `re_log_types` from here so even if we used `arrow2_convert` we'd have to duplicate the definition of `Tuid` in here, which would have the same issue.

We could come up with some crazy `include_str!` tricks but I don't think that's worth the complexity; the real nice fix would be to register custom formatters at compile time (https://github.com/rerun-io/rerun/issues/1775).

Reply:
Oh right `Tuid` doesn't live with the rest of them, yeah i can import that one